test/regress/regress-28.txt

```diff
@@ -15,6 +15,5 @@ section(CODE) {
   }
 }
 (;; STDERR ;;;
-out/test/regress/regress-28/regress-28.wasm:000001c: error: type mismatch at end of function, expected [] but got [any]
-000001c: error: EndFunctionBody callback failed
+0000019: error: function body shorter than given size
```

Comment:
Why did we loose the filename here?

Reply:
Hmm, nice catch. :-/ It turns out that right now _every_ error that's caught inside the `BinaryReader` itself is getting saved without a filename, because the `BinaryReader` doesn't know anything about filenames and when it creates an `Error`, it sets the error's `Location` where only the `offset` is set (the `filename` field is empty).

We can change this if you want -- e.g.
- we could make `BinaryReader` take an optional filename in its constructor that's just used for error logging, or
- we could make BinaryReaderIR/BinaryReaderInterp::OnError method modify the error's Location before saving it, i.e. from their own `GetLocation()` function (which produces a richer Location with the filename filled in)

I just tried this and it ended up changing a ton of the test diagnostic output, but it's definitely doable either way.

Hope you're okay if we merge this now and can revisit the filenames in error logging in a future PR.

Reply:
yeah I'm not too bothered about that.